### ISSUE TITLE
Provice notify signal connect example. 

### ISSUE THREAD

**Issue by gavr123456789** [StefanSalewski/gintro#171]
I see a line "And the notify signals for gobject properties like "notify::cursor-moved" for entry widgets should work now" in the description, but I cant find any example of notify signal.  
  
Rigth now I have an error:  
```nim
proc selectFileCb(self: ToggleButton, pspec: ParamSpec, row: FileRow ) = ...
...
toggleBtn.connect("notify::active", selectFileCb, row)

```
Error:  
```
Error: type mismatch: got <ToggleButton, FileRow>
but expected one of: 
proc selectFileCb(self: ToggleButton; pspec: ParamSpec; row: FileRow)
  first type mismatch at position: 2
  required type for pspec: ParamSpec
  but expression 'cast[FileRow](xdata)' is of type: FileRow

expression: selectFileCb(cast[gtk4.ToggleButton](h), cast[FileRow](xdata))
```  
  
And I can't understand why the error is on the pspec argument if the second argument in the callback is pspec.

**Comment by StefanSalewski** [comment]
Example:

```
salewski@nuc ~ $ grep notify ~/SDT/sdt.nim 
  entry.connect("notify::cursor-position", entryNotify, result)
```

Will investigate your issue in more detail later.

**Comment by StefanSalewski** [comment]
Your error is strange indeed. Have you imported glib? I have seen strange errors when I forget that in some cases.

If you can not solve it, please post a short minimal example so that I can try to compile it.

**Comment by StefanSalewski** [comment]
Maybe the problem is that the active property of ToggleButton is of type bool, which I have not tested yet.

But I wonder why you need a notify for that property at all. We have

https://docs.gtk.org/gtk4/signal.Button.clicked.html

https://docs.gtk.org/gtk4/method.ToggleButton.get_active.html

That should be enough.

**Comment by gavr123456789** [comment]
> Have you imported glib  
  
Yes, imported `import gintro/[gtk4, gobject, gio, adw, glib]`, ParamSpec in gobject

**Comment by gavr123456789** [comment]
Now all works

**Comment by gavr123456789** [comment]
> But I wonder why you need a notify for that property at all  
  
Thats a interesting part. I have a list of files that represented as toggle button widgets:  
![image](https://user-images.githubusercontent.com/30507409/129160714-0f7dbdef-9b72-4c2a-bc29-52c25c90017a.png)
Toggled means that file selected. I am currently working on the implementation of select all. I realized that I can't get all the widgets from the ListView to mark activate their ToggleButtons. So I decided to switch from NoSelection to MultiSelection Model. But it turned out, I can't do two-way binding on `ListItem.selected` because it is read only https://docs.gtk.org/gtk4/property.ListItem.selected.html. And I need the old functionality wort the same, that is, clicking on the Toggle button should activate the selected ItemList, and vice versa, if the ItemList is selected, I need the button to light up.  
  
By doing the following `discard listitem.bindProperty("selected", row.btn1, "active", {syncCreate})` I found that simply setting a property in activated does not cause a toggled signal, which is tied to the functionality of opening file CRUD Revealer. That's why I needed notify to trigger the signal not by toggled, but by changing activate, which now changes from the fact that the ItemList has become selected.
